src/main/java/io/sqreen/sasdk/backend/AuthConfig.java

```diff
@@ -0,0 +1,36 @@
+package io.sqreen.sasdk.backend;
+
+import com.google.common.collect.ImmutableMultimap;
+
+public class AuthConfig {
```

Comment:
I find this class hardly an improvement over the previous one. You've replaced a clear public interface:

```
    Optional<String> getAPIKey();	
    Optional<String> getAppName();	
    Optional<String> getSessionKey();
}
```

with this:

```
    public ImmutableMultimap<String, String> getAllProps();
```

This is certainly not easier to reason about. Not only does the client of the class lose information about what the valid configuration settings are, they also unnecessarily have to deal with an intermediate map. The class is also more clumsy to mock. Instead of:

```groovy
def conf = mock(AuthConfig)
conf.appName.retuns(Optional.of('myapp'))
```

we need
```groovy
def conf = mock(Authconfig)
def mockMap = mock(ImmutableMap)
conf.allProps.returns(mockMap)
mockMap.get('appName').returns('myapp')
```

and this test is also my fragile because the interaction with the map can happen through other map functions.